File: src/config/llm.js
/**
 * Grok (xAI) LLM Configuration
 */
module.exports = {
  apiKey: process.env.GROK_API_KEY,
  apiUrl: process.env.GROK_API_URL || 'https://api.x.ai/v1',
  model: process.env.GROK_MODEL || 'grok-3-mini-fast',
  embeddingModel: process.env.GROK_EMBEDDING_MODEL || 'grok-embedding-public',
  maxTokens: parseInt(process.env.MAX_TOKENS, 10) || 1024,
  temperature: parseFloat(process.env.TEMPERATURE) || 0.1, // low = less hallucination
  retryAttempts: 3,
  retryDelay: 1000 // ms between retries
};

File: app/config/llm.py
"""
LLM Configuration (Groq — OpenAI-compatible API)
Embeddings are handled locally via ONNX runtime (no API needed).
"""
import os

API_KEY = os.getenv("GROQ_API_KEY", "")
API_URL = os.getenv("GROQ_API_URL", "https://api.groq.com/openai/v1")
MODEL = os.getenv("GROQ_MODEL", "llama-3.3-70b-versatile")

# Local embedding model (ONNX runtime — runs locally, free)
EMBEDDING_MODEL = os.getenv("EMBEDDING_MODEL", "all-MiniLM-L6-v2")

TEMPERATURE = float(os.getenv("TEMPERATURE", "0.1"))
MAX_TOKENS = int(os.getenv("MAX_TOKENS", "2048"))
RETRY_ATTEMPTS = int(os.getenv("RETRY_ATTEMPTS", "3"))
RETRY_DELAY = float(os.getenv("RETRY_DELAY", "1.0"))  # seconds
